pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -512,11 +515,24 @@ func Convert_v1_VirtualMachine_To_api_Domain(vmi *v1.VirtualMachineInstance, dom
 
 	currentAutoThread := defaultIOThread
 	currentDedicatedThread := uint(autoThreads + 1)
+
+	var numQueues *uint
+	if (vmi.Spec.Domain.Devices.BlockMultiQueue != nil) && (*vmi.Spec.Domain.Devices.BlockMultiQueue) {
```

Comment:
Ok, so if we don't know how many cpus we got, we simply disable multi-queue. Is this intentional, in the context that a VMI can run on all node-cores when cpu assignment is best-effort?

Reply:
As far as I understand it, if the number of CPU's is unspecified at this point, then the number of CPU's will be 1 (since this is the domain XML conversion step). With that in mind, yes it was entirely intentional to disable multi-queue since the number of queues should match the number of CPU's--and one queue is the same as mutli-queue disabled.

Reply:
> As far as I understand it, if the number of CPU's is unspecified at this point, then the number of CPU's will be 1 (since this is the domain XML conversion step)

The number of requested CPU resources is not related to how many cores the VMI has. If the VMI would be the only workload on the node and there is no limit on the VMI set, it can actually all CPU cores until the kubelet or the OOM killer comes in (best-effort QoS without load on the node).

> With that in mind, yes it was entirely intentional to disable multi-queue since the number of queues should match the number of CPU's--and one queue is the same as mutli-queue disabled.

With my comment above, is that still true?

Reply:
First off, thanks for clarifying. However, I'm unsure how the code should respond in this case. Since we don't have any guarantee of how many physical CPU's would be allocated, I don't think any assumption would be valid in this case.

I'm leaning towards a single queue still being the right call. If we ask for more queues and the system eventually comes under load, we don't have a way to change how many queues were requested later.